spring-integration-core/src/main/java/org/springframework/integration/support/leader/LockRegistryLeaderInitiator.java

```diff
@@ -168,6 +175,14 @@ public LockRegistryLeaderInitiator(LockRegistry locks, Candidate candidate) {
 		this.candidate = candidate;
 	}
 
+	@Override
+	public void afterPropertiesSet() throws Exception {
```

Comment:
We don't need this method here. We can just simply have that 
```
private ExecutorService executorService =
			Executors.newSingleThreadExecutor(new CustomizableThreadFactory("lock-leadership-"));
```
 as before.

Indeed we need some `boolean` property to control external executor. But in other places we call it like `executorServiceExplicitlySet` - much cleaner and consistent.
You set it to `true` in the setter

Reply:
OK, I'll go back to that.  I just personally don't like setter doing anything other than setting the target variable.  Also I thought `afterPropertiesSet()` would given more chance for extension and would also not create an `ExecutorService` & `ThreadFactory` when not needed.